pkg/collector/corechecks/network-devices/cisco-sdwan/payload/cedge_interface.go

```diff
@@ -0,0 +1,126 @@
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/corechecks/network-devices/cisco-sdwan/client"
+	devicemetadata "github.com/DataDog/datadog-agent/pkg/networkdevice/metadata"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// https://github.com/YangModels/yang/blob/9442dda17a9a5f1f0db548512446e3d9ca37a955/vendor/cisco/xe/17131/Cisco-IOS-XE-interfaces-oper.yang#L305
+var cEdgeOperStatusMap = map[string]devicemetadata.IfOperStatus{
+	"if-oper-state-down":             devicemetadata.OperStatusDown,
+	"if-oper-state-invalid":          devicemetadata.OperStatusDown,
+	"if-oper-state-ready":            devicemetadata.OperStatusUp,
+	"if-oper-state-no-pass":          devicemetadata.OperStatusDown,
+	"if-oper-state-test":             devicemetadata.OperStatusTesting,
+	"if-oper-state-unknown":          devicemetadata.OperStatusUnknown,
+	"if-oper-state-dormant":          devicemetadata.OperStatusDormant,
+	"if-oper-state-not-present":      devicemetadata.OperStatusNotPresent,
+	"if-oper-state-lower-layer-down": devicemetadata.OperStatusLowerLayerDown,
+}
+
+// https://github.com/YangModels/yang/blob/9442dda17a9a5f1f0db548512446e3d9ca37a955/vendor/cisco/xe/17131/Cisco-IOS-XE-interfaces-oper.yang#L212
+var cEdgeAdminStatusMap = map[string]devicemetadata.IfAdminStatus{
+	"if-state-unknown": devicemetadata.AdminStatusDown,
+	"if-state-up":      devicemetadata.AdminStatusUp,
+	"if-state-down":    devicemetadata.AdminStatusDown,
+	"if-state-test":    devicemetadata.AdminStatusTesting,
+}
+
+// CEdgeInterface is an implementation of CiscoInterface for cEdge devices
+type CEdgeInterface struct {
+	client.CEdgeInterfaceState
+}
+
+// ID returns an unique interface ID
+func (itf *CEdgeInterface) ID() string {
+	return fmt.Sprintf("%s:%s", itf.VmanageSystemIP, itf.Ifname)
+}
+
+// Index returns the interface index
+func (itf *CEdgeInterface) Index() int {
+	index, err := strconv.Atoi(itf.Ifindex)
+	if err != nil {
+		log.Warnf("Unable to parse cEdge interface %s index %s", itf.Ifname, itf.Ifindex)
+	}
+	return index
+}
+
+// Speed returns the interface speed
+func (itf *CEdgeInterface) Speed() int {
+	speed, err := strconv.Atoi(itf.SpeedMbps)
+	if err != nil {
+		log.Warnf("Unable to parse cEdge interface %s speed %s", itf.Ifname, itf.SpeedMbps)
+	}
+	return speed
+}
+
+// OperStatus returns the interface oper status
+func (itf *CEdgeInterface) OperStatus() devicemetadata.IfOperStatus {
+	return convertOperStatus(cEdgeOperStatusMap, itf.IfOperStatus)
+}
+
+// AdminStatus returns the interface admin status
+func (itf *CEdgeInterface) AdminStatus() devicemetadata.IfAdminStatus {
+	return convertAdminStatus(cEdgeAdminStatusMap, itf.IfAdminStatus)
+}
+
+// Metadata returns the interface metadata
+func (itf *CEdgeInterface) Metadata(namespace string) devicemetadata.InterfaceMetadata {
+	index, err := strconv.Atoi(itf.Ifindex)
+	if err != nil {
+		log.Warnf("Unable to parse cEdge interface %s index %s", itf.Ifname, itf.Ifindex)
+	}
```

Comment:
We continue processing after this error? This may end up in sending duplicates because `index` will be zero here

I'm not sure if the `payload` package is the right place to log things for the agent, could it make sense to do it one level higher with the code that actually uses it?

Reply:
It felt weird to drop a whole interface only because we can't find it's index (The vManage uses the interface name as interface identifier for a given device), but yeah on second though this will not play well with the backend, good catch!

Reply:
Fixed in 9347cb0